staging/src/k8s.io/kubectl/pkg/cmd/apply/apply_test.go

```diff
@@ -1382,3 +1383,28 @@ func TestForceApply(t *testing.T) {
 		})
 	}
 }
+
+func TestRunApplyWithPodGeneratedName(t *testing.T) {
+	expectedError := "error: from testing-: cannot use generate name with apply"
+	cmdutil.BehaviorOnFatal(func(str string, code int) {
+		panic(str)
```

Comment:
Going via `panic()` seems a bit complicated - why not just check the error here, similar to https://github.com/kubernetes/kubernetes/pull/88998/files#diff-9e758ba495ba77ca7c7c1419d305959bR1221

Reply:
Change done :)